components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java

```diff
@@ -182,9 +189,17 @@ public static String getOrganizationId() {
      *
      * @return the username of the authenticated user.
      */
-    public static String getAuthenticatedUsername() {
-
-        return PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+    public static String getAuthenticatedUsername() throws OrganizationManagementException {
+
+        String username = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (username == null) {
+            Optional<User> maybeUser = organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
+                    getUserId(), getOrganizationId());
+            if (maybeUser.isPresent()) {
+                return maybeUser.get().getUsername();
+            }
```

Comment:
```suggestion
            return organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
                    getUserId(), getOrganizationId()).map(User::getUsername).orElse(username);
```

Reply:
Fixed by - https://github.com/wso2-extensions/identity-organization-management-core/pull/10/commits/185a6f96da864aa5b64c7adcf2d49ba8a7957c6c
`map(User::getUsername).orElse(username)`. => `orElse(null)`. Set as this based on IDE warning as username is always null.